### PR TITLE
rg-190: Add drag'n'drop to template editor

### PR DESCRIPTION
**Pull request by honkster-nazarii**
Demo

https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-resumegemm/assets/133093930/e3a305fb-5120-4dcb-93c6-84578021726a

### CODE REVIEW

**Comment by honkster-nazarii** [comment]
[Add drag'n'drop to template editor](https://trello.com/c/3IERB4k3/190-add-dragndrop-to-template-editor)

**Comment by honkster-nazarii** [comment]
checking for performance rn

**Review by Valentyn13 — APPROVED** [review]

**Review by siefimov — APPROVED** [review]

**Review by MaxPoplavskiy — APPROVED** [review]

**Review by RevolveStrm — APPROVED** [review]

File: frontend/src/bundles/edit-template/components/template-block/template-block.tsx [hunk 1]
```diff
@@ -1,15 +1,121 @@
...
+
     return (
-        <div key={id} style={styles}>
-            {items.map((item) => (
-                <TemplateItem key={item.id} {...item} />
-            ))}
+        <div
+            ref={setNodeRef}
+            style={{
+                opacity: isDragging ? 0.5 : 1,
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/edit-template/components/template-block/template-block.tsx:89]: This condition occurs twice, put it in a function
Consider using classes for such cases

File: frontend/src/bundles/edit-template/components/template-editor/template-editor.tsx [hunk 2]
```diff
@@ -1,19 +1,95 @@
...
+    templateSettings: TemplateSettings;
+    setTemplateSettings: React.Dispatch<React.SetStateAction<TemplateSettings>>;
 };
 
-const TemplateEditor: React.FC<Properties> = ({ settings }) => {
+const TemplateEditor: React.FC<Properties> = ({
+    templateSettings,
+    setTemplateSettings,
+}) => {
+    const [active, setActive] = useState<Active | null>(null);
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/edit-template/components/template-editor/template-editor.tsx:30]: Try to come up with better name

File: frontend/src/bundles/edit-template/components/template-item/template-item.tsx [hunk 3]
```diff
@@ -1,40 +1,73 @@
...
+
     return (
-        <div key={id} style={styles}>
-            {handleType(tagName)}
+        <div
+            className={templateItemStyles.template_item}
+            key={id}
+            style={{
+                zIndex: isDragging ? 1000 : 0,
+                opacity: isDragging ? 0.5 : 1,
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/edit-template/components/template-item/template-item.tsx:52]: Same as above

File: frontend/src/bundles/edit-template/context/template-context.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { createContext } from 'react';
+
+import { type TemplateSettings } from '../types/types';
+
+const TemplateContext = createContext<{
+    templateSettings: TemplateSettings;
+    setTemplateSettings: React.Dispatch<React.SetStateAction<TemplateSettings>>;
+}>(
+    {} as {
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/edit-template/context/template-context.ts:9]: Is there a way to avoid using `as`?

**Review by Vitaliy-Kalenichenko — CHANGES_REQUESTED** [review]

**Pull request merged**